x-pack/plugins/observability_solution/infra/server/routes/metadata/lib/get_metric_metadata.ts

```diff
@@ -74,11 +76,31 @@ export const getMetricMetadata = async (
     },
   };
 
+  if (nodeType === 'host') {
+    set(metricQuery, 'body.aggs', {
+      monitoredHost: {
+        filter: getFilterByIntegration(SYSTEM_INTEGRATION),
+        aggs: {
+          name: {
+            terms: {
+              field: HOST_NAME_FIELD,
+              size: 1,
+              order: {
+                _key: 'asc',
+              },
+            },
+          },
+        },
+      },
```

Comment:
Q and nit: Do we need lodash for this?

We could do this directly on the `metricQuery`

```suggestion
...(nodeType === 'host'
          ? {
              monitoredHost: {
                filter: getFilterByIntegration(SYSTEM_INTEGRATION),
                aggs: {
                  name: {
                    terms: {
                      field: HOST_NAME_FIELD,
                      size: 1,
                      order: {
                        _key: 'asc',
                      },
                    },
                  },
                },
              },
            }
          : undefined),
```

Reply:
I replaced it with a `getHasDataFromSystemIntegration` as discussed